src/hotspot/share/prims/jvmtiTrace.cpp

```diff
@@ -277,15 +277,10 @@ const char *JvmtiTrace::safe_get_thread_name(Thread *thread) {
   if (!thread->is_Java_thread()) {
     return thread->name();
   }
-  oop threadObj = thread->as_Java_thread()->threadObj();
-  if (threadObj == NULL) {
-    return "NULL";
-  }
-  oop name = java_lang_Thread::name(threadObj);
-  if (name == NULL) {
-    return "<NOT FILLED IN>";
-  }
-  return java_lang_String::as_utf8_string(name);
+
+  // If the target JavaThread is not protected, then we return the
+  // specified non-NULL string:
+  return thread->as_Java_thread()->get_thread_name("<NOT FILLED IN>");
```

Comment:
That's not what the original logic is doing. We only return "NOT FILLED IN" if we find that the java.lang.Thread oop does not have a name set. That's not normally possible but this code is intended to be 100% safe, hence the check. (And it may be possible to see null during the attach process ...)

I would expect the target thread to already be protected when this method is called in any case. So while this code partially duplicates JavaThread::get_thread_name() I'm not convinced we need to change this code to use JavaThread::get_thread_name() instead.

Reply:
My intent was to have safe_get_thread_name() take advantage of
JavaThread::get_thread_name() when we know that we're dealing
with a JavaThread. Shared code means less maintenance, but it
does look like I didn't achieve equivalence so I'm going to back
out my changes for JvmtiTrace::safe_get_thread_name().